keel-web/src/main/kotlin/com/netflix/spinnaker/config/KeelConfigurationFinalizer.kt

```diff
@@ -96,6 +98,9 @@ class KeelConfigurationFinalizer(
         log.info("Registering Constraint Attributes sub-type {}: {}", attributeType.name, attributeType.type.simpleName)
         extensionRegistry.register(attributeType.type, attributeType.name)
       }
+
+    extensionRegistry.register(DependsOnConstraintAttributes::class.java, "depends-on")
```

Comment:
I assume this is to support deserializing these objects from the database?

Reply:
yeah, unfortunately. The way to 'automatically' do this is right above, but they need to be a part of a stateless evaluator. I felt this was easier but grosser.

Reply:
Couldn't you just remove the filter above and check if `attributeType` is not null?